src/test/compile-fail/issue-14853.rs

```diff
@@ -20,7 +20,7 @@ struct X { data: u32 }
 
 impl Something for X {
     fn yay<T: Str>(_:Option<X>, thing: &[T]) {
-    //~^ ERROR the requirement `T: Str` appears on the impl method
+    //~^ ERROR E0276
```

Comment:
Are all of these scenarios covered by the ui tests?

 * If so, should we just delete these compile-fail tests?
 * And if not, then I as a test reviewer personally prefer keeping the longer messages, as it gives me a better chance of divining what the original intent was from the test author.